File: AEDS/aeds2/tp1/palindromo/c/palindromo.c
#include <stdio.h>
#include <stdbool.h>

// funcao para determinar tamanho da palavra
int str_len(char *palavra){
    int length = 0;
	while(*(palavra+length)){ 
		length++;
	}
	return(length);
}

//funcao para comparar strings 
bool str_cmp(char palavra1[], char palavra2[]){
    bool resp = true;
    int length = str_len(palavra1);

    if(length != str_len(palavra2)){
        resp = false;    
    } else {
        for(int i = 0; i < length; i++){
            if(palavra1[i] != palavra2[i]){
               resp = false;
               i = length;
            } 
        }
    }
    return (resp);
}

//procedimento para determinar se uma palavra e' ou nao palindromo
void classificarPalindromo(char palavra[]){
    int n = str_len(palavra);
    bool resp = true;
    for(int i = 0; i < n / 2; i++){
        if(palavra[i] != palavra[n - i - 1]){
           resp = false;
           i = n;
        } 
    }
    if(resp) printf("SIM\n");
    else printf("NAO\n");
}

//metodo principal
int main (){
    char palavra[500];
    scanf(" %[^\n]", palavra);  
    
    while(!str_cmp(palavra, "FIM")){
        classificarPalindromo(palavra);
        scanf(" %[^\n]", palavra);  
    }
    return 0;
}

File: 2° semestre/aeds2/tp1/palindromo/Palindromo.java
import java.util.Scanner;

public class Palindromo {
    //funcao para comparar strings e determinar se sao identicas
    public static boolean isEquals(String palavra1, String palavra2){
        int n = palavra1.length();
        boolean resp = true;

        if(n != palavra2.length()){
            resp = false;
        } else {
            for(int i = 0; i < n; i++){
                if(palavra1.charAt(i) != palavra2.charAt(i)){
                    resp = false;
                    i = n;
                 }
            }
        }
        return (resp);
    }
    // metodo para classificar se uma palavra e' ou nao um palindromo
    public static void classificarPalindromo(String palavra){
        int n = palavra.length();
        boolean resp = true;
        
        for(int i = 0; i < n / 2; i++){
            if(palavra.charAt(i) != palavra.charAt(n - i - 1)){
                resp = false;
                i = n;
            }
        }
        if(resp) System.out.println("SIM");
        else System.out.println("NAO");
    }
    // metodo principal
    public static void main (String[] args){
        String palavra;
        Scanner entrada = new Scanner(System.in);
        
        palavra = entrada.nextLine();
       
        while(!isEquals(palavra, "FIM")){
            classificarPalindromo(palavra);
            palavra = entrada.nextLine();
        }
        entrada.close();
    }
}